contrib/config/apollo/apollo.go

```diff
@@ -89,18 +106,101 @@ func NewSource(opts ...Option) config.Source {
 	if err != nil {
 		panic(err)
 	}
-	return &apollo{client}
+
+	return &apollo{client: client, opt: &op}
+}
+
+// genKey got the key of config.KeyValue pair.
+// eg: namespace.ext with subKey got namespace.subKey
+func genKey(ns, sub string) string {
```

Comment:
may need some unit tests for these tool function, as well as `converProperties` and `format` functions